src/core/BoxGeometry.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef CORE_BOX_GEOMETRY_HPP
+#define CORE_BOX_GEOMETRY_HPP
+
+#include <utils/Vector.hpp>
+
+#include <bitset>
+#include <cassert>
+
+class BoxGeometry {
+public:
+  /** Flags for all three dimensions whether pbc are applied (default). */
+  std::bitset<3> m_periodic = 0b111;
+  /** Side lengths of the box */
+  Utils::Vector3d m_length = {1, 1, 1};
+
+  /**
+   * @brief Set periodicity for direction
+   *
+   * @param coord The coordinate to set the periodicity for.
+   * @param val True if this direction should be periodic.
+   */
+  void set_periodic(unsigned coord, bool val) { m_periodic.set(coord, val); }
+
+  /**
+   * @brief Check periodicity in direction.
+   *
+   * @param coord Direction to check
+   * @return true iff periodic in direction.
```

Comment:
Comment refers to this line.

Reply:
https://en.wikipedia.org/wiki/If_and_only_if :-)

Reply:
Iff this was indeed intentional, leave as is ;-)